variables.tf

```diff
@@ -2,8 +2,60 @@
 # Input Variables
 ########################################################################################################################
 
-#variable "my_variable" {
-#  type        = string
-#  description = "A description of my variable"
-#  default     = "default_value"
-#}
+variable "resource_group_id" {
+  description = "The Id of an existing IBM Cloud resource group where the watsonx Assistant instance will be grouped."
+  type        = string
+}
+
+variable "region" {
+  description = "IBM Cloud region where watsonx Assistant instance will be created."
+  type        = string
+}
+
+variable "tags" {
+  description = "Metadata labels describing this watsonx Assistant instance."
+  type        = list(string)
+  default     = []
+}
+
+variable "watsonx_assistant_name" {
+  description = "The name of the watsonx assistant instance."
+  type        = string
+}
+
+variable "existing_assistant_instance" {
+  description = "CRN of the an existing watsonx Assistant instance."
+  type        = string
+  default     = null
+}
+
+variable "watsonx_assistant_plan" {
+  description = "The plan that is required to provision the watsonx Assistant instance."
+  type        = string
+  default     = "trial"
+  validation {
+    condition = anytrue([
+      var.watsonx_assistant_plan == "do not install",
+      var.watsonx_assistant_plan == "trial",
+      var.watsonx_assistant_plan == "lite",
+      var.watsonx_assistant_plan == "plus",
+      var.watsonx_assistant_plan == "enterprise",
+      var.watsonx_assistant_plan == "enterprisedataisolation",
+    ])
+    error_message = "You must use a Trial, Lite, Plus, Enterprise, or Enterprise with data isolation plan. [Learn more](https://cloud.ibm.com/docs/watson-assistant?topic=watson-assistant-admin-managing-plan)."
+  }
+  validation {
+    condition     = !(contains(["trial", "lite"], var.watsonx_assistant_plan) && var.watsonx_assistant_service_endpoints != "public")
+    error_message = "The 'Trial' and 'Lite' plans only support public endpoints."
+  }
+}
+
+variable "watsonx_assistant_service_endpoints" {
+  description = "The type of service endpoints. Possible values: 'public', 'private', 'public-and-private'."
+  type        = string
+  default     = "public"
+  validation {
+    condition     = contains(["public", "public-and-private", "private"], var.watsonx_assistant_service_endpoints)
```

Comment:
you you sure private only is supported?

Reply:
Yes, the dropdown shows three options and this endpoint option is not shown for `trial` and `lite` plans. 
<img width="1080" alt="image" src="https://github.com/user-attachments/assets/eaf806be-f5ea-466a-a22c-a64d4c6666ce">

Reply:
do you mean endpoints cant be selected for trial and lite plan? will it be always public?
We should add a validation on endpoints as per the plan if any.

Reply:
@ocofaigh @maheshwarishikha - Please check [this](https://github.ibm.com/GoldenEye/issues/issues/11380#issuecomment-98189101). The values for Lite and Trial are `null` which I believe should be `public`. However [asked](https://ibm-cloudplatform.slack.com/archives/C05REAEBN6T/p1732449204230919) in the channel for clarity.